Load in test and training CSV files from Week 3

In [ ]:
import pandas as pd
import numpy as np
from pathlib import Path
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score

#load in the cleaned test and training datasets from week 3
test_df = pd.read_csv("cleaned_test.csv")
train_df = pd.read_csv("cleaned_training.csv")

print(f"Test DataFrame shape: {test_df.shape}")
print(f"Training DataFrame shape: {train_df.shape}")

feature_cols = ['LivingArea','BedroomsTotal','BathroomsTotalInteger',
                      'LotSizeSquareFeet', 'zip_median_price', 'city_median_price' ]

X_train = train_df[feature_cols].copy()
y_train = train_df['ClosePrice'].copy()

X_test = test_df[feature_cols].copy()
y_test = test_df['ClosePrice'].copy()

print(f"X_train shape: {X_train.shape} | y_train shape: {y_train.shape}")
print(f"X_test shape:  {X_test.shape}  | y_test shape:  {y_test.shape}")

Test DataFrame shape: (11969, 7)
Training DataFrame shape: (70538, 7)
X_train shape: (70538, 6) | y_train shape: (70538,)
X_test shape:  (11969, 6)  | y_test shape:  (11969,)


In [ ]:
print("Training missing values:")
print(X_train.isnull().sum())

print("\nTest missing values:")
print(X_test.isnull().sum())

Training missing values:
LivingArea               0
BedroomsTotal            0
BathroomsTotalInteger    0
LotSizeSquareFeet        0
zip_median_price         0
city_median_price        0
dtype: int64

Test missing values:
LivingArea               0
BedroomsTotal            0
BathroomsTotalInteger    0
LotSizeSquareFeet        0
zip_median_price         0
city_median_price        0
dtype: int64


In [ ]:
import numpy as np

model = LinearRegression()
model.fit(X_train, y_train)

y_train_pred = model.predict(X_train)
y_test_pred = model.predict(X_test)

train_results = r2_score(y_train, y_train_pred)
test_results = r2_score(y_test, y_test_pred)
print('Linear Regression Results: \n')
print(f'Training R^2: {train_results:.4f}')
print(f'Test R^2: {test_results:.4f}')

Linear Regression Results: 

Training R^2: 0.7778
Test R^2: 0.7611


Results:

The model had an R² score of 0.7611 on the test set, indicating that it explains approximately 76% of the variation in home sale prices. Both R² results are similar to each other with a difference of 0.0167 representing that the model is generalizing well without overfitting.